test/python/test_ibmqjob_states.py

```diff
@@ -30,132 +31,139 @@ def setUp(self):
 
     def test_unrecognized_status(self):
         job = self.run_with_api(UnknownStatusAPI())
-        self.assertStatus(job, JobStatus.INITIALIZING)
-        self.wait_for_initialization(job)
-        self.assertIsInstance(job.exception, IBMQJobError)
-        self.assertStatus(job, JobStatus.ERROR)
+        with self.assertRaises(JobError):
+            self.wait_for_initialization(job)
 
     def test_validating_job(self):
         job = self.run_with_api(ValidatingAPI())
-        self.assertStatus(job, JobStatus.INITIALIZING)
+        self.assertEqual(job.status(), JobStatus.INITIALIZING)
 
         self.wait_for_initialization(job)
-        self.assertStatus(job, JobStatus.VALIDATING)
-
-        self._current_api.progress()
+        self.assertEqual(job.status(), JobStatus.VALIDATING)
 
     def test_error_while_creating_job(self):
         job = self.run_with_api(ErrorWhileCreatingAPI())
-        self.assertStatus(job, JobStatus.INITIALIZING)
+        self.assertEqual(job.status(), JobStatus.INITIALIZING)
 
         self.wait_for_initialization(job)
-        self.assertStatus(job, JobStatus.ERROR)
+        self.assertEqual(job.status(), JobStatus.ERROR)
 
     def test_error_while_running_job(self):
         job = self.run_with_api(ErrorWhileRunningAPI())
-        self.assertStatus(job, JobStatus.INITIALIZING)
+        self.assertEqual(job.status(), JobStatus.INITIALIZING)
 
         self.wait_for_initialization(job)
-        self.assertStatus(job, JobStatus.RUNNING)
+        self.assertEqual(job.status(), JobStatus.RUNNING)
 
         self._current_api.progress()
-        self.assertStatus(job, JobStatus.ERROR)
+        self.assertEqual(job.status(), JobStatus.ERROR)
+
+        # We have an error message this time
```

Comment:
Remove the comment and the blank line, please.

Reply:
I may change the comment, because I wanted to state that an error message is not always present when an error occurs, but this test is sending one so we capture it.

Reply:
Removing the white line though